kairon/async_callback/auth.py

```
@@ -0,0 +1,37 @@
+        if not header.startswith("Bearer "):
+            raise AuthError("Bad Authorization header")
+
+        token = header.split(" ", 1)[1]
+        try:
+            decoded = jwt.decode(token, SECRET_KEY, algorithms=[ALGORITHM])
+            claims  = Authentication.decrypt_token_claims(decoded["sub"])
+            if claims.get("type") != TOKEN_TYPE.DYNAMIC.value:
+                raise AuthError("Invalid token type")
+
+        except ExpiredSignatureError:
+            raise AuthError("Token expired")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)